ssl/record/rec_layer_s3.c

```diff
@@ -368,7 +368,8 @@ int ssl3_write_bytes(SSL *s, int type, const void *buf_, int len)
      * promptly send beyond the end of the users buffer ... so we trap and
      * report the error in a way the user will notice
      */
-    if ((unsigned int)len < s->rlayer.wnum) {
+    if (((unsigned int)len < s->rlayer.wnum) 
+          || ((tot != 0) && ((unsigned int)len < ((unsigned int)tot + s->rlayer.wnum)))) {
```

Comment:
Did you really mean `wnum` here? Just above here we set `tot = s->rlayer.wnum`, so really what you are checking is `len < 2 * s->rlayer.wnum` - which I don't think is what you intended?